libs/api/domains/cms/src/lib/search/contentful.service.ts

```diff
@@ -233,22 +233,33 @@ export class ContentfulService {
     // In case of delta updates, we need to resolve embedded entries to their root model
     if (syncType !== 'full' && nestedEntries) {
       logger.info('Finding root entries from nestedEntries')
-      const alreadyProcessedIds = items.map((entry) => entry.sys.id)
-      for (const entryId of nestedEntries) {
-        // Due to the limitation of Contentful Sync API, we need to query every entry one at a time
-        // with regular sync, triggered by a webhook, these calls 1 - 2 at most
-        const linkedEntries = await this.limiter.schedule(() => {
-          return this.linksToEntry(entryId, locale)
-        })
-        linkedEntries.forEach((entry) => {
-          // No need to import the same document twice
-          if (
-            !alreadyProcessedIds.includes(entry.sys.id) &&
-            environment.indexableTypes.includes(entry.sys.contentType.sys.id)
-          ) {
-            items.push(entry)
-          }
-        })
+
+      // For now we will look for linked entries up to depth 2
+      for (let i = 1; i <= 2; i++) {
```

Comment:
It would be optimal to have this go to the same depth as the max include depth in our Contentful service `contentful.service.ts` (this is 10 atm), to prevent the partial sync issues completely.
Did you look into this in regards to added time/pressure when running the sync?

Reply:
The sync endpoint gets really slow as the number of nested entries to look up increases. I tested with ~60 entries and the sync took around 30 seconds. I think it's fine because there is usually only 1 entry when Contentful calls the webhook.

But increasing the depth might be risky because we don't know how fast the number grows with depth, best case would be `1 -> 2 -> 3 -> 4` but there's no guarantee it won't be `1 -> 10 -> 100 -> 1000`

The queries for these pages only go to depth 2 so I think this is safe for now, although not the best solution.

Reply:
I see your point, true the safe approach is to limit this to current max query depth.
I think this is 3 in our queries atm set by menu: `Grouped menu->menus->menuLinks->childLinks`
That should do until we can apply a more permanent solution to this issue

Reply:
I've changed it from 2 to 3.